cob_twist_controller/include/cob_twist_controller/inverse_differential_kinematics_solver.h

```diff
@@ -58,33 +58,27 @@ class InverseDifferentialKinematicsSolver
      * kinematics for
      *
      */
-        InverseDifferentialKinematicsSolver(const KDL::Chain& chain, CallbackDataMediator& data_mediator) :
+    InverseDifferentialKinematicsSolver(const KDL::Chain& chain, CallbackDataMediator& data_mediator) :
         chain_(chain),
         jac_(chain_.getNrOfJoints()),
         jnt2jac_(chain_),
         callback_data_mediator_(data_mediator),
         constraint_solver_factory_(data_mediator, jnt2jac_)
     {
-            last_p_in_vec_ = t_Vector6d::Zero();
+        last_p_in_vec_ = t_Vector6d::Zero();
+        this->kinematic_extension_.reset(KinematicExtensionBuilder::create_extension(this->params_));
     }
 
     virtual ~InverseDifferentialKinematicsSolver() {};
     
-    /** CartToJnt for chain using SVD including base and various DampingMethods **/
+    /** CartToJnt for chain using SVD considering KinematicExtensions and various DampingMethods **/
     virtual int CartToJnt(const JointStates& joint_states,
                           const KDL::Twist& v_in,
-                          const KDL::Frame &base_position,
-                          const KDL::Frame &chain_base,
                           KDL::JntArray& qdot_out);
 
-    inline virtual int CartToJnt(const JointStates& joint_states,
-                                 const KDL::Twist& v_in,
-                                 KDL::JntArray& qdot_out)
+    inline void SetInvDiffKinSolverParams(InvDiffKinSolverParams params)
```

Comment:
Must be removed and must not be used anymore!!! 

Instead of the method resetAll must be called to ensure reconstruction of solvers etc.